lib/puppet/type/private_key.rb

```diff
@@ -0,0 +1,66 @@
+Puppet::Type.newtype(:private_key) do
+  desc 'manage a private key'
```

Comment:
one thing, that I find confusing as a user: we already have a `privkey` type:

https://github.com/theforeman/puppet-certs/blob/af1c91414f5f4c98cafb78180b0a340db7e3c52a/lib/puppet/type/privkey.rb#L3-L4

but it does something totally different, albeit having a similar name?